internal/pkg/template/workload.go

```diff
@@ -456,6 +491,17 @@ type NetworkLoadBalancer struct {
 	CertificateRequired bool
 }
 
+// ApplicationLoadBalancer holds configuration that's needed for an Application Load Balancer.
+type ApplicationLoadBalancer struct {
+	PublicSubnetCIDRs []string
+	Listener          []ApplicationLoadBalancerRoutineRule
```

Comment:
I believe these are more equivalent to listener rules:
```suggestion
	Rules          []ALBListenerRule
```